librz/bin/format/elf/elf_symbols.c

```diff
@@ -216,6 +216,11 @@ static size_t get_number_of_symbols_from_gnu_hash(ELFOBJ *bin) {
 	ut32 number_of_bucket = BREAD32(bin->b, pos);
 	ut32 symbol_base = BREAD32(bin->b, pos);
 	ut32 bitmask_nwords = BREAD32(bin->b, pos);
+
+	if (number_of_bucket == UT32_MAX || symbol_base == UT32_MAX || bitmask_nwords == UT32_MAX) {
```

Comment:
shouldn't this be compared with the file size instead with its max?

Reply:
The error value is UT32_MAX.

Reply:
In addition to checking the error you might want to check if they are "valid" values as well. I guess that's what @wargio means.

Reply:
what i mean is that i think you should check the plausibility of having for example `0xefefefef` as `number_of_bucket ` in an elf file of `32 kb` for example.

Reply:
something like

if (number_of_bucket * sizeof(ut32) + offset > bin->size) {
	return 0
}

Reply:
yes exactly